Lista.h

```diff
@@ -126,10 +130,36 @@ void Lista<Dato>::delDato(unsigned pos)
     tam--;
 }
 
+template<class Dato>
+unsigned Lista<Dato>::getPosicion(Dato d)
+{
+    bool estaEnLista  = false;
+    unsigned i = 1 ,posicion ;
+
+    while( i <= this->getTam() || !estaEnLista)
+    {
+        Dato datoLista = this->getDato(i);
+
+        if(datoLista == d)
+        {
+            estaEnLista = true;
+            posicion = i;
+        }
+
+        i++;
+    }
+
+    if(!estaEnLista)
+        posicion = -1;
```

Comment:
- No hace falta usar el método `getTam()` porque es un atributo de la clase, podés poner directamente `this->tam`
- Se puede sacar el bloque `if(!estaEnLista)` e inicializar la variable en `-1` directamente. Si el dato existe, cuando se llegue a la posición se va a modificar el valor, y sino sigue siendo `-1`
- En el `while` la condicion deberia ser `&&`. Con el `||` sigue recorriendo hasta el final, aunque `estaEnLista` sea `true`, porque necesita que una de las dos condiciones se cumpla para iterar
```suggestion
    bool estaEnLista  = false;
    unsigned i = 1, posicion = -1;
    while( i <= this->tam && !estaEnLista)
    {
        Dato datoLista = this->getDato(i);
        if(datoLista == d)
        {
            estaEnLista = true;
            posicion = i;
        }
        i++;
    }
```
@AlexTeper99 cuidado con esto cuando hagas el grafo, porque si el dato no existe va a devolver `-1`. Vas a tener que chequear que el valor obtenido sea distinto de `-1` o mayor a cero antes de intentar obtener un valor de la matriz, porque sino va a tirar error

Reply:
Oki

Reply:
Lo mismo que dije abajo. El metodo getDato, podria llamarse obtenerDato.